### DISCUSSION TITLE
When viewing a deleted-file diff, how do I open the file as it existed before the deletion in a normal (non-diff) editor?

### DISCUSSION THREAD

**Discussion by ELLIOTTCABLE** [gitkraken/vscode-gitlens#2626] [Q&A]
According to [this comment](https://github.com/gitkraken/vscode-gitlens/issues/646#issuecomment-462139682), there's an "open at revision ...",

> <@eamodio > Ah, so to just open a readonly version of a specific revision use, Open Revision... — opens a version of the active file with whatever revision you choose from the quick pick.

… but that doesn't seem to exist, that I can find:

<img width="626" alt="image" src="https://user-images.githubusercontent.com/200/231243762-a71d291b-6aa2-4f43-93b9-b87ab96a5ea7.png">

Is it perhaps unavailable in deleted-file diffs? How can I get the previously-existing form of that file, from `main` or whatever, opened in a readonly/historical virtual editor?

**Comment by jkelroy** [comment]
Hello @ELLIOTTCABLE, if you select the file diff and then use the command palette to search for revision, do you see the expected options after? 

I noticed that there are no options if the diff is not focused and have written up an issue here. https://github.com/gitkraken/vscode-gitlens/issues/2627

**Reply by ELLIOTTCABLE** [reply]
The diff-window is focused in the above screenshot. If it's relevant, I use this in GitHub-Issues-and-Pull-Requests 'review mode?'

<img width="1651" alt="Screenshot 2023-04-11 at 2 59 07 PM" src="https://user-images.githubusercontent.com/200/231287334-6133cfef-7d0b-4976-ac93-c17102028d35.png">

**Reply by jkelroy** [reply]
As a note, you can simply look at the [file history](https://help.gitkraken.com/gitlens/side-bar/#file-history-view) to see the previous versions of this file.

As for the issue with you not seeing this option, while in GitHub Issues and Pull Requests, I am still able to see the revisions options so it sounds like you may be seeing a separate issue than the previously mentioned one. Do you see any options when using `command + shift + P` and then searching GitLens? What version of GitLens and VS Code are you using?

![image](https://user-images.githubusercontent.com/27739784/231585296-7e58e89d-90b1-4cda-baf4-5b8ffd0978b3.png)
